android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -136,43 +145,71 @@ protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean
                         }
                     }
                 }
-                return MPUrl.getUrl(builder.build().toString(), defaultUrl);
+                return MPUrl.getUrl(builder.build().toString(), !isDefaultUrl ? generateDefaultURL(builder.build(), defaultDomain, (SERVICE_VERSION_4 + "/" + pathPostfix)) : null);
             case EVENTS:
                 subdirectory = overridesSubdirectory ? "" : SERVICE_VERSION_2 + "/";
+                pathPostfix = mApiKey + "/events";
                 uri = new Uri.Builder()
                         .scheme(BuildConfig.SCHEME)
                         .encodedAuthority(url)
-                        .path(subdirectory + mApiKey + "/events")
+                        .path(subdirectory + pathPostfix)
                         .build();
-                return MPUrl.getUrl(uri.toString(), defaultUrl);
+
+                return MPUrl.getUrl(uri.toString(), !isDefaultUrl ? generateDefaultURL(uri, defaultDomain, (SERVICE_VERSION_2 + "/" + pathPostfix)) : null);
             case ALIAS:
                 subdirectory = overridesSubdirectory ? "" : SERVICE_VERSION_1 + "/identity/";
+                pathPostfix = mApiKey + "/alias";
                 uri = new Uri.Builder()
                         .scheme(BuildConfig.SCHEME)
                         .encodedAuthority(url)
-                        .path(subdirectory + mApiKey + "/alias")
+                        .path(subdirectory + pathPostfix)
                         .build();
-                return MPUrl.getUrl(uri.toString(), defaultUrl);
+                return MPUrl.getUrl(uri.toString(), !isDefaultUrl ? generateDefaultURL(uri, defaultDomain, (SERVICE_VERSION_1 + "/identity/" + pathPostfix)) : null);
             case IDENTITY:
                 subdirectory = overridesSubdirectory ? "" : SERVICE_VERSION_1 + "/";
+                pathPostfix = identityPath;
                 uri = new Uri.Builder()
                         .scheme(BuildConfig.SCHEME)
                         .encodedAuthority(url)
                         .path(subdirectory + identityPath)
```

Comment:
to be consistent with everywhere else, maybe this should be 
```suggestion
                        .path(subdirectory + pastPostfix)
```